azure/scope/managedcontrolplane_test.go

```diff
@@ -1038,3 +1041,171 @@ func TestAreLocalAccountsDisabled(t *testing.T) {
+				},
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "my-cluster",
+						Namespace: "dummy-ns",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "my-cluster",
+						Namespace: "dummy-ns",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						SubscriptionID: "00000000-0000-0000-0000-000000000001",
+						VirtualNetwork: infrav1.ManagedControlPlaneVirtualNetwork{
+							ResourceGroup: "dummy-rg",
+							Name:          "vnet1",
+							Subnet: infrav1.ManagedControlPlaneSubnet{
+								Name: "subnet1",
+								PrivateEndpoints: infrav1.PrivateEndpoints{
+									{
+										Name:     "my-private-endpoint",
+										Location: "westus2",
+										PrivateLinkServiceConnections: []infrav1.PrivateLinkServiceConnection{
+											{
+												Name:                 "my-pls-connection",
+												PrivateLinkServiceID: "my-pls-id",
+												GroupIDs: []string{
+													"my-group-id-1",
+												},
+												RequestMessage: "my-request-message",
+											},
+										},
+										CustomNetworkInterfaceName: "my-custom-nic",
+										PrivateIPAddresses: []string{
+											"IP1",
+											"IP2",
+										},
+										ApplicationSecurityGroups: []string{
+											"ASG1",
+											"ASG2",
+										},
+										ManualApproval: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Expected: []azure.ASOResourceSpecGetter[*asonetworkv1.PrivateEndpoint]{
+				&privateendpoints.PrivateEndpointSpec{
+					Name:                       "my-private-endpoint",
+					Namespace:                  "dummy-ns",
+					ResourceGroup:              "dummy-rg",
+					Location:                   "westus2",
+					CustomNetworkInterfaceName: "my-custom-nic",
+					PrivateIPAddresses: []string{
+						"IP1",
+						"IP2",
+					},
+					SubnetID: "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/dummy-rg/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
+					ApplicationSecurityGroups: []string{
+						"ASG1",
+						"ASG2",
+					},
+					ClusterName: "my-cluster",
+					PrivateLinkServiceConnections: []privateendpoints.PrivateLinkServiceConnection{
+						{
+							Name:                 "my-pls-connection",
+							RequestMessage:       "my-request-message",
+							PrivateLinkServiceID: "my-pls-id",
+							GroupIDs: []string{
+								"my-group-id-1",
+							},
+						},
+					},
+					ManualApproval: true,
+					AdditionalTags: make(infrav1.Tags, 0),
+				},
+			},
+		},
+	}
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			g := NewWithT(t)
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(c.Input.ControlPlane).Build()
+			c.Input.Client = fakeClient
+			s, err := NewManagedControlPlaneScope(context.TODO(), c.Input)
+			g.Expect(err).To(Succeed())
```

Comment:
Do we really need to build a client to test this function? Or could we get around having to call `NewManagedControlPlaneScope` by defining a `*ManagedControlPlaneScope` directly?

Reply:
Good catch, we don't need to. Updating..

Reply:
Added via 178d32f4039b9117767f49c599edcaf0bb1c202a